LandSandBoat/server
# [lua] [sql] Removed animated_weapon.lua mixin, all references to the mixin, move fragment drops to SQL

Co-authored-by: CatsEyeXI <beneliezer@gmail.com>



**_I affirm:_**

## What does this pull request do?

Removes **animated_weapon.lua** mixin and all references to it from the .lua files of the Animated Weapon mobs.  This PR also removes the code from the .lua files of the Animated Weapon mobs that dictated which fragment each Animated Weapon drops.

PR https://github.com/LandSandBoat/server/pull/3516 (authored by CatsEyeXI) (PR closed because was combined into this PR) moves the relevant 100% chance fragment drops out of the mob .lua files and into SQL where they should belong going forward.

Reference the BG-Wiki pages for the Animated Weapons (https://www.bg-wiki.com/ffxi/Category:Animated_Weapon#Event_Appearances) to see that all list their relevant fragment as a Guaranteed drop.  FFXIDB can also be referenced to display the drop rates of the fragments but the rates may be skewed as the data may contain old school dynamis drop rate data as well as drop rates reported by those that did not kill the 15 demon NMs before killing an Animated Weapon.

History:

**animated_weapon.lua** mixin was an outdated relic of the 75 cap Dynamis era where the killing of the 15 demon NMs on the 5 towers in Dynamis-Xarcabard was required before the Animated Weapons dropped their relevant fragment. The mixin, along with the previously removed function **activateAnimatedWeapon** (removed in https://github.com/LandSandBoat/server/commit/56e5f4d70424a755968e4ed7fe99e046709f586a) are what caused the animation sub-state changes of the the Animated Weapons to allow for them to drop their fragment once all 15 demon NMs were killed.

Neo Dynamis now only requires the relevant goad item to be obtained to spawn the relevant animated weapon. Once the relevant animated weapon is defeated, the relevant fragment is a **G**uaranteed 100% drop.  

## Steps to test these changes

Force spawn any of the Animated Weapon NMs in Dynamis - Xarcabard **BEFORE** these changes are merged, notice that no fragment drops off of the Animated Weapon where it is a guaranteed drop.

Force spawn any of the Animated Weapon NMs in Dynamis - Xarcabard **AFTER** these changes are merged, notice that the correct fragment will always drop off the Animated Weapon like it should.
